### PR TITLE
Add recurring task feature

### LINKED ISSUE #91: Add recurring tasks functionality
Able to make some events repeat for a certain period (weekly lectures, weekly meetings, etc)

### PR DESCRIPTION
**Pull request by lerxcl**
Done implementation + unit tests + user documentation.
Left with developer guide.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -363,11 +370,11 @@ Action | Format, Examples
 **List (Contacts and Tasks)** | `list all`
 **List Contacts** | `list contact`
 **Help** | `help`
-**Add To Do** | `add todo desc/DESCRIPTION date/DATE time/TIME` <br> e.g., `add todo desc/finish user guide draft date/09/08/2020 time/2300`
+**Add To Do** | `add todo desc/DESCRIPTION date/DATE time/TIME [recurring/VALUE UNIT]` <br> e.g., `add todo desc/finish user guide draft date/09/08/2020 time/2300`
 **List To Dos** | `list todo`
 **Remove To Do** | `delete todo INDEX` <br> e.g., `delete todo 3`
 **Show To Do** | `show todo INDEX` <br> e.g., `show todo 3`
-**Add Event** | `add event desc/DESCRIPTION startdate/DATE starttime/TIME enddate/DATE endtime/TIME` <br> e.g., `add event desc/meeting startdate/12-12-2020 starttime/1000 enddate/12-12-2020 endtime/1130`
+**Add Event** | `add event desc/DESCRIPTION startdate/DATE starttime/TIME enddate/DATE endtime/TIME [recurring/VALUE UNIT]` <br> e.g., `add event desc/meeting startdate/12-12-2020 starttime/1000 enddate/12-12-2020 endtime/1130`
```
Comment (kevnw) [inline docs/UserGuide.md]: Can you give a proper example? The example here does not seem to use the recurring feature.

File: src/main/java/seedu/address/logic/commands/DoneCommand.java [hunk 2]
```diff
@@ -31,12 +30,23 @@ public CommandResult execute(Model model) throws CommandException {
...
         if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
         }
 
         Task taskToMark = lastShownList.get(targetIndex.getZeroBased());
-        model.markAsDone(taskToMark);
+        AddCommand recurrenceAddCommand = model.markAsDone(taskToMark);
+        if (recurrenceAddCommand != null) {
+            System.out.println("adding again");
```
Comment (kevnw) [inline src/main/java/seedu/address/logic/commands/DoneCommand.java]: Remove sysout commands.

File: src/test/java/seedu/address/logic/commands/DoneCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
+package seedu.address.logic.commands;
```
Comment (kevnw) [inline src/test/java/seedu/address/logic/commands/DoneCommandTest.java:1]: Good Job!

**Review by kevnw — CHANGES_REQUESTED** [review]
Just some minor nitpick. Fix some parts and it LGTM! 👍

File: docs/UserGuide.md (hunk 1 above)
Comment (lerxcl) [reply docs/UserGuide.md]: Oh yea I didn't update the bottom example, will do so!

File: src/main/java/seedu/address/logic/commands/DoneCommand.java (hunk 2 above)
Comment (lerxcl) [reply src/main/java/seedu/address/logic/commands/DoneCommand.java]: Okay noted!

File: src/test/java/seedu/address/logic/commands/DoneCommandTest.java (hunk 3 above)
Comment (lerxcl) [reply src/test/java/seedu/address/logic/commands/DoneCommandTest.java:1]: Thanks :smile:

**Review by kevnw — APPROVED** [review]
LGTM! 👍

**Pull request merged**